metadata/src/utils/validation.rs

```diff
@@ -798,6 +798,25 @@ mod tests {
         assert_ne!(aba_hash, abb_hash);
     }
 
+    #[test]
+    fn hash_cache_gets_filled_with_correct_hashes() {
+        let mut registry = Registry::new();
+        let a_type_id = registry.register_type(&meta_type::<A>()).id;
+        let b_type_id = registry.register_type(&meta_type::<B>()).id;
+        let registry: PortableRegistry = registry.into();
+
+        let mut cache = HashMap::new();
+
+        let a_hash = get_type_hash(&registry, a_type_id, &mut cache);
+        let b_hash = get_type_hash(&registry, b_type_id, &mut cache);
+
+        let CachedHash::Hash(a_cache_hash) = cache[&a_type_id] else { panic!() };
+        let CachedHash::Hash(b_cache_hash) = cache[&b_type_id] else { panic!() };
+
+        assert_eq!(a_hash, a_cache_hash);
+        assert_eq!(b_hash, b_cache_hash);
```

Comment:
Can we also check that `a_hash == b_hash` :)

Reply:
I don't think we want the hashes to be the same, because `A` and `B` are different structs, with different fields:
```
struct A { pub b: Box<B> }
struct B { pub a: Box<A> }
```
Maybe I understood you wrong, what did you mean?

Reply:
Ah, sorry yeah, I mean, it'd be nice to have a test like:

```rust
let a_type_id = registry.register_type(&meta_type::<A>()).id;

let a_hash1 = get_type_hash(&registry, a_type_id, &mut cache);
let a_hash2 = get_type_hash(&registry, a_type_id, &mut cache);

assert_eq!(a_hash1, a_hash2);
```

ie, we want to check that sharing the cache across different uses doesn't change the hashes that are produced for some type.